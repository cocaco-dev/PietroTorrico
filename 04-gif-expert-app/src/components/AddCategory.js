import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); //Undefined si no se pone las comillas
    const handleInputChange = (e) => {        
        setInputValue(e.target.value)        
    }
    const handleSubmit = (e) => {
        e.preventDefault();   
        console.log('ENVIO', inputValue);     
        if(inputValue.trim().length > 2){
            setCategories(cate => [inputValue, ...cate]); //cate hace referencia al contenido del componente  
            setInputValue('');
        }
    }

    return (
        <form onSubmit = {handleSubmit}>        
            {/* <p>{inputValue}</p>     */}
            <input
                type="text"
                value = {inputValue}
                onChange={handleInputChange}
                placeholder = 'Ingresa una categoria'
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}