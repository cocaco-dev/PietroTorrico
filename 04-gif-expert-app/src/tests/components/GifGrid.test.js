import React from 'react';
import { useFectchGifs } from '../../hooks/useFetchGifs';
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas con el componente <GifGrid/>', ()=> {
    
    const category = 'One PunchMan';   

    test('Debemos probar el snapshot ', () => {        

        useFectchGifs.mockReturnValue({ //ERROR, NO ES RECONOCIDO COMO UNA FUNCIÓN
            data: [],
            loading: false
        });
        const wrapper = shallow(<GifGrid category = {category}/>)
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar items cuando se cargan imágenes UseFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'Titulo de ejemplo'
        },
        {
            id: '123',
            url: 'https://google.com',
            title: 'Titulo de ejemplo'
        }
        ]

        useFectchGifs.mockReturnValue({ 
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category = {category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    

});