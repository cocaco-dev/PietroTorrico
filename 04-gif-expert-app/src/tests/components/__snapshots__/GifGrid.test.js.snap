// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con el componente <GifGrid/> Debe de mostrar items cuando se cargan im√°genes UseFetchGifs 1`] = `
<Fragment>
  <h3>
    One PunchMan
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo de ejemplo"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Titulo de ejemplo"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`pruebas con el componente <GifGrid/> Debemos probar el snapshot  1`] = `
<Fragment>
  <h3>
    One PunchMan
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
